### PR TITLE
Chapter 6

### PR DESCRIPTION
**Pull request by mirulzaei**

### CODE REVIEW

**Comment by mirulzaei** [comment]
ready

File: app/helpers/application_helper.rb [hunk 1]
```diff
@@ -1,2 +1,6 @@
 module ApplicationHelper
+  def full_title page_title = ""
```
Comment (ngocvt-0484) [inline app/helpers/application_helper.rb]:
```suggestion
  def full_title page_title
```

File: app/helpers/application_helper.rb [hunk 2]
```diff
@@ -1,2 +1,6 @@
...
+    base_title = "Sample App"
```
Comment (ngocvt-0484) [inline app/helpers/application_helper.rb:3]: use i18n

File: app/models/user.rb [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  before_save :downcase_email
+
+  private
+
+  def downcase_email
+    email.downcase!
+  end
+end
+
+end
```
Comment (ngocvt-0484) [inline app/models/user.rb:23]: remove "end" because redundant

**Comment by mirulzaei** [comment]
ready

File: app/assets/stylesheets/application.scss [hunk 4]
```diff
@@ -13,3 +13,6 @@
  *= require_tree .
  *= require_self
  */
+ @import 'bootstrap-sprockets';
+ @import 'bootstrap';
+ @import 'color';
```
Comment (chungpv-1008) [inline app/assets/stylesheets/application.scss:18]: please add endline for this file

File: config/routes.rb [hunk 5]
```diff
@@ -1,8 +1,8 @@
 Rails.application.routes.draw do
-    root "static_pages#home"
-    get "/home", to: "static_pages#home"
-    get "/help", to: "static_pages#help"
-    get "/about", to: "static_pages#about"
-    get "/contact", to: "static_pages#contact"
-    get "/signup", to: "users#new"
+  root "static_pages#home" 
+  get "/home", to: "static_pages#home"
```
Comment (chungpv-1008) [inline config/routes.rb]:
```suggestion
  root "static_pages#home"
  
  get "/home", to: "static_pages#home"
```

File: config/settings.yml [hunk 6]
```diff
@@ -9,3 +9,12 @@ links:
...
     michael_hartl: https://www.michaelhartl.com/
+
+user:
+  name:
+    max_length: 50
+  email:
+    max_length: 255
+    regex: !ruby/regexp '/\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i'
+  password:
+    min_length: 6
```
Comment (chungpv-1008) [inline config/settings.yml]: add endline for this file

File: config/routes.rb (hunk 5 above)
Comment (mirulzaei) [reply config/routes.rb]: done

File: config/settings.yml (hunk 6 above)
Comment (mirulzaei) [reply config/settings.yml]: done

**Comment by mirulzaei** [comment]
ready

**Comment by vanvtt-0952** [comment]
merged

**Pull request merged**